internal/pkg/cli/task_run.go

```diff
@@ -371,6 +379,42 @@ func (o *runTaskOpts) Validate() error {
 		}
 	}
 
+	if len(o.secrets) > 0 {
+		if err := o.validateSecrets(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSecrets() error {
+	if o.appName == "" && o.env == "" && !o.acknowledgeSecretsAccess {
```

Comment:
On line [382](https://github.com/aws/copilot-cli/blob/524658bf37f5a8b5fb4eeaecfa519b9105605111/internal/pkg/cli/task_run.go#L382) I am checking for `o.secrets > 0` so I guess we don't need to check for `o.secrets` again. 

Also  `o.acknowledgeSecretsAccess` is the only one extra condition that I have appended to `o.appName == "" && o.env == ""`. 
Is that okay? or you think I should separate `o.acknowledgeSecretsAccess` condition as well?

Reply:
It's no biggie whether `o.acknowledgeSecretsAccess` is in the same condition statement as app and env! I'm fine with both as long as it's obvious for any future readers what the conditional means.

The more important point is to flip the conditional and return early so that it improves code readability.

Reply:
Sorry for adding more on top of this 😛 I think we should also take `--execution-role` into consideration. Like we probably  shouldn't need to prompt the question if `--execution-role` is specified.

Reply:
maybe something like this?

```go
if o.executionRole != "" {
  return nil
}
if appName != "" || envName != "" {
  return nil
}
if !o.acknowledgeSecretsAccess {
  // log to prompt them to specify either acknowledge flag or execution role if they want to proceed
  return error
}
// log and ask questions
...
```
I think one benefit for this is it would make it crystal clear in what different situation we don't prompt the confirmation question and we'll have less indented code block